File: 643_max_avg_subarray_1/max_avg_subarray_1.py
class Solution:
    def findMaxAverage(self, nums: List[int], k: int) -> float:
        sum1 = sum(nums[:k])
        max_sum = sum1

        for i in range(k,len(nums)):
            sum1 -= nums[i-k]
            sum1 += nums[i]
            if(sum1 > max_sum):
                max_sum = sum1

        return float(max_sum / k)

File: 643_max_avg_subarray_1/max_avg_subarray_1.cpp
class Solution {
public:
    double findMaxAverage(vector<int>& nums, int k) {
        int sum1 = 0.0;
        int i;
        for(i = 0; i < k; i++) {
            sum1 += nums[i];
        }
        int max_sum = sum1;
        for(i = k; i < nums.size(); i++) {
            sum1 += nums[i] - nums[i - k];
            if(sum1 > max_sum)
                max_sum = sum1;
        }
        return double(max_sum) / k;
    }
};
